spec/app/domain/authentication/authn-azure/application_identity_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnAzure::ApplicationIdentity do
+  include_context "azure setup"
+
+  let(:user_assigned_identity_service_id_scoped_annotation) { double("UserAssignedIdentityAnnotation") }
+  let(:system_assigned_identity_service_id_scoped_annotation) { double("UserAssignedIdentityServiceIdScopeAnnotation") }
+
+  before(:each) do
+    define_host_annotation(host_annotation_type: user_assigned_identity_service_id_scoped_annotation, host_annotation_key: "#{granular_annotation_type}/user-assigned-identity", host_annotation_value: "some-user-assigned-service-id-scoped-value")
+    define_host_annotation(host_annotation_type: system_assigned_identity_service_id_scoped_annotation, host_annotation_key: "#{granular_annotation_type}/system-assigned-identity", host_annotation_value: "some-system-assigned-service-id-scoped-value")
+
+    define_host_annotation(host_annotation_type: non_azure_annotation, host_annotation_key: "#{global_annotation_type}/non-azure-annotation", host_annotation_value: "some-non-azure-value")
+  end
+
+  context "An application identity in annotations" do
+    subject(:application_identity) {
+      Authentication::AuthnAzure::ApplicationIdentity.new(
+        role_annotations: role_annotations,
+        service_id:       test_service_id
+      )
+    }
+    context("with a global scoped constraint") do
+      let(:role_annotations) { [subscription_id_annotation, resource_group_annotation, user_assigned_identity_annotation, system_assigned_identity_annotation] }
+
+      it "Returns Hash of the constraint and its value" do
```

Comment:
i don't really like the phrase `returns`. The subject doesn't return the hash. The correct phrase should be "has a constraints hash with..." or something like that.